refactor(dev-infra): cleanup shared package dependencies

devversion opened the pull request:
* See individual commits

devversion committed:
refactor(dev-infra): use shelljs instead of fs-extra for benchmark utils

We added a new dependency on `fs-extra` to the dev-infra package. We can
remove this dependency and replace it with `shelljs` that is extensively
used in other places already.

The motiviation is that we can reduce dependencies needed for
for consumption of the shared dev-infra package.

josephperrott commented on dev-infra/tmpl-package.json:
```diff
@@ -21,34 +22,30 @@
+    "typescript": "<from-root>"
   },
   "peerDependenciesMeta": {
+    "@bazel/buildifier": {
```
@bazel/buildifier has been removed from `peerDependencies`? It should still be present correct?

devversion replied:
yeah, that was by mistake. Just wiped all `@bazel/*` dependencies.

devversion committed:
refactor(dev-infra): cleanup shared package dependencies

Cleans up the dependencies used in the shared dev-infra package
configuration. With the recent benchmarking utilities that have
been added, a lot of peer dependencies have been added.

We decided that we don't want to list every used dependencies as
peer dependency as that could result in unnecessary churn/noise
for consumers of the dev-infra package. Additionally, not all parts
of the dev-infra package are necessarily used.

Due to this, we want to apply the following rules for the package
dependencies:

1. If a dependency is only used in a shipped Bazel macro/rule that can be
optionally consumed, omit it from `package.json`. Bazel reports the
missing dependency on its own, so we want to avoid adding it to the
package json file.

2. Otherwise, if the dependency is large and commonly used (like
buildifier), add it to the `peerDependencies`. If not, add it
to the dependencies that are always brought in. We consider it
as acceptable to bring in a few small dependencies that might not
be used or not. Making all of those option would complicate the
use of the dev-infra package.

ds

josephperrott reviewed (approved):
LGTM
